include/boost/lexical_cast/detail/converter_lexical_streams.hpp

```diff
@@ -82,10 +82,10 @@
 #endif
 
 #if ( defined(_MSC_VER) && _MSC_VER < 1900 ) || ( defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR) )
```

Comment:
Unconditionally using `snprintf` was merged, but breaks support to [MinGW so was rolled back](https://github.com/boostorg/lexical_cast/pull/51#issuecomment-1236049443). This same snippet is used in Boost.Assert, and now Boost.Core so should be safe.

Reply:
it appears that no one investigated why it failed, just "Well, it failed so roll back" in spite of @biodranik's assertion that "Mingw sources definitely have snprintf".

Reply:
@pdimov will have a better answer than I since he is the original author. I am transplanting this snippet where needed because Apple Clang explodes with warnings when building or testing Boost.